libmpv/csharp/Form1.cs

```diff
@@ -0,0 +1,167 @@
+        private MpvSetOption _mpvSetOption;
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private delegate int MpvSetOptionString(IntPtr mpvHandle, byte[] name, byte[] value);
+        private MpvSetOptionString _mpvSetOptionString;
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private delegate int MpvGetPropertystring(IntPtr mpvHandle, byte[] name, int format, ref IntPtr data);
+        private MpvGetPropertystring _mpvGetPropertyString;
+
+        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
+        private delegate int MpvSetProperty(IntPtr mpvHandle, byte[] name, int format, ref byte[] data);
+        private MpvSetProperty _mpvSetProperty;
+
+        public Form1()
+        {
+            InitializeComponent();
+        }
+
+        private object GetDllType(Type type, string name)
+        {
+            IntPtr address = GetProcAddress(_libMpvDll, name);
+            if (address != IntPtr.Zero)
+                return Marshal.GetDelegateForFunctionPointer(address, type);
+            return null;
+        }
+
+        private void LoadMpvDynamic()
+        {
+            _libMpvDll = LoadLibrary("mpv-1.dll"); // The dll is included in the DEV builds by lachs0r: https://mpv.srsfckn.biz/
+            _mpvCreate = (MpvCreate)GetDllType(typeof(MpvCreate), "mpv_create");
+            _mpvInitialize = (MpvInitialize)GetDllType(typeof(MpvInitialize), "mpv_initialize");
+            _mpvTerminateDestroy = (MpvTerminateDestroy)GetDllType(typeof(MpvTerminateDestroy), "mpv_terminate_destroy");
+            _mpvCommand = (MpvCommand)GetDllType(typeof(MpvCommand), "mpv_command");
+            _mpvSetOption = (MpvSetOption)GetDllType(typeof(MpvSetOption), "mpv_set_option");
+            _mpvSetOptionString = (MpvSetOptionString)GetDllType(typeof(MpvSetOptionString), "mpv_set_option_string");
+            _mpvGetPropertyString = (MpvGetPropertystring)GetDllType(typeof(MpvGetPropertystring), "mpv_get_property");
+            _mpvSetProperty = (MpvSetProperty)GetDllType(typeof(MpvSetProperty), "mpv_set_property");
+        }
+
+        public void Pause()
+        {
+            if (_mpvHandle == IntPtr.Zero)
+                return;
+
+            var bytes = Encoding.UTF8.GetBytes("yes\0");
+            _mpvSetProperty(_mpvHandle, Encoding.UTF8.GetBytes("pause\0"), MpvFormatString, ref bytes);
+        }
+
+        private void Play()
+        {
+            if (_mpvHandle == IntPtr.Zero)
+                return;
+
+            var bytes = Encoding.UTF8.GetBytes("no\0");
+            _mpvSetProperty(_mpvHandle, Encoding.UTF8.GetBytes("pause\0"), MpvFormatString, ref bytes);
+        }
+
+        public bool IsPaused()
+        {
+            if (_mpvHandle == IntPtr.Zero)
+                return true;
+
+            IntPtr lpBuffer = Marshal.AllocHGlobal(10);
+            _mpvGetPropertyString(_mpvHandle, Encoding.UTF8.GetBytes("pause\0"), MpvFormatString, ref lpBuffer);
```

Comment:
There is a memory leak here. ``mpv_get_property(MPV_FORMAT_STRING)`` sets *data to a pointer to an mpv-allocated string that must be freed with ``mpv_free()``. In C, the function would be used like this:

```c
char *str;
mpv_get_property(mpv, "pause", MPV_FORMAT_STRING, &str);
/* Do stuff with the string */
mpv_free(str);
```

In this code, the HGlobal isn't used, since mpv will overwrite the pointer to it with its own allocated memory. Also, neither the HGlobal or mpv's allocated string are freed.